corehq/apps/domain/forms.py

```diff
@@ -537,6 +537,9 @@ def __init__(self, *args, **kwargs):
             # if the cloudcare_releases flag was just defaulted, don't bother showing
             # this setting at all
             del self.fields['cloudcare_releases']
+        if self.project.default_geocoder_location == 'default' \
+                or not domain_has_privilege(self.domain, privileges.CLOUDCARE):
+            del self.fields['default_geocoder_location']
```

Comment:
I think you also need to update the code in `DomainGlobalSettingsForm.__init__` that's based on the index of different fields: https://github.com/dimagi/commcare-hq/blob/64d433902599bbfa56261047ff3d4a8cdf9af38f/corehq/apps/domain/forms.py#L392-L393

I think that `__init__` code is what's causing the logo field to not appear, as discussed in the USH Tech chat on Skype.

Reply:
Changed the indexes and tested. This fixed the upload logo in my local